### PR TITLE
Cat fixes

### PR DESCRIPTION
**Pull request by harmonyn**
- Space cats now have sprites for lying and sitting. Fixes https://github.com/tgstation/-tg-station/issues/15607. Fixes https://github.com/tgstation/-tg-station/issues/15890.
- Player-controlled cats now have a Rest verb. No more cats doing the worm.
- Adds more flavor text for cats' Life(). I feel like these are happening way more often than I thought they would, but on the other hand that is 100% characteristic of cats. But with the family persistence update I can see how it can get spammy fast. Thoughts?

### CODE REVIEW

**Comment by CosmicScientist** [comment]
D'aaaw. :+1:

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 1]
```diff
@@ -124,20 +136,23 @@
 /mob/living/simple_animal/pet/cat/Life()
 	if(!stat && !buckled)
 		if(prob(1))
-			emote("me", 1, pick("stretches out for a belly rub.", "wags its tail."))
+			emote("me", 1, pick("stretches out for a belly rub.", "wags its tail.", "lies down."))
 			icon_state = "[icon_living]_rest"
 			resting = 1
+			canmove = 0
```
Comment (tkdrg) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: these should be calls to `update_canmove()` instead of manually setting this

**Comment by tkdrg** [comment]
you did it the wrong way around, this is how you'll break it by overriding the icon_states
undo your newest commit and do an if(client) inside your update_canmove override
```
/mob/living/simple_animal/pet/cat/update_canmove()
	..()
	if(client)
		if(resting)
			icon_state = "[icon_living]_rest"
		else
			icon_state = "[icon_living]"
```

**Comment by bgobandit** [comment]
@tkdrg you don't seem to understand this code at all

the icon states are supposed to update for ALL cat movements. player-controlled or not. adding a check for client there means that the icons will not update when cats lie down due to Life(), which is how it will happen 99% of the time

meanwhile, the Life() is not supposed to happen if there is a player controlling the cat. otherwise you get player-controlled cats randomly deciding to lie down, which is stupid. hence, the client check.

I have tested this thoroughly and it works

**Comment by tkdrg** [comment]
Then you can keep both checks, but your code is still wrong. In the current version, the following is what happens:
1) Life() rolls, cat gets icon_state = "[icon_living]_sit"
2) Life() sets resting = 1 and calls update_canmove()
3) update_canmove() sees resting is 1 and sets icon_state = "[icon_living]_rest"
How is this so hard to understand

**Comment by bgobandit** [comment]
@tkdrg

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 2]
```diff
@@ -30,6 +30,18 @@
...
+/mob/living/simple_animal/pet/cat/New()
+	..()
+	verbs += /mob/living/proc/lay_down
+
+/mob/living/simple_animal/pet/cat/update_canmove()
+	..()
+	if(client && resting)
+		icon_state = "[icon_living]_rest"
+	else
+		icon_state = "[icon_living]"
```
Comment (duncathan) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: This else clause means none of the other icon states will be seen

Comment (bgobandit) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: oops I kind of got lazy due to spending fucking forever on the meme conflict

**Comment by bgobandit** [comment]
@tkdrg

can someone merge this so the catsplosion can get fixed

**Pull request merged**